examples/guestbook/README.md

```diff
@@ -1,203 +1,253 @@
-
-Use (or just create) the file `examples/guestbook/redis-master-controller.json` which describes a single [pod](../../docs/pods.md) running a redis key-value server in a container:
-
-Note that, although the redis server runs just with a single replica, we use [replication controller](../../docs/replication-controller.md) to enforce that exactly one pod keeps running (e.g. in a event of node going down, the replication controller will ensure that the redis master gets restarted on a healthy node).   This could result in data loss.
-
-
-```js
-{
-   "kind":"ReplicationController",
-   "apiVersion":"v1",
-   "metadata":{
-      "name":"redis-master",
-      "labels":{
-         "name":"redis-master"
-      }
-   },
-   "spec":{
-      "replicas":1,
-      "selector":{
-         "name":"redis-master"
-      },
-      "template":{
-         "metadata":{
-            "labels":{
-               "name":"redis-master"
-            }
-         },
-         "spec":{
-            "containers":[
-               {
-                  "name":"master",
-                  "image":"redis",
-                  "ports":[
-                     {
-                        "containerPort":6379
-                     }
-                  ]
-               }
-            ]
-         }
-      }
-   }
-}
+This example requires a running Kubernetes cluster.  See the [Getting Started guides](../../docs/getting-started-guides) for how to get started. As noted above, if you have a GKE cluster set up, go [here](https://cloud.google.com/container-engine/docs/tutorials/guestbook) instead.
+
+### Step One: Start up the redis master
+
+**Note**: The redis master in this example is *not* highly available.  Making it highly available would be an interesting, but intricate exercise— redis doesn't actually support multi-master deployments at this point in time, so high availability would be a somewhat tricky thing to implement, and might involve periodic serialization to disk, and so on.
+
+To start the redis master, use the file `examples/guestbook/redis-master-controller.yaml`, which describes a single [pod](../../docs/pods.md) running a redis key-value server in a container.
+
+Although we have a single instance of our redis master, we are using a [replication controller](../../docs/replication-controller.md) to enforce that exactly one pod keeps running. E.g., if the node were to go down, the replication controller will ensure that the redis master gets restarted on a healthy node. (In our simplified example, this could result in data loss.)
+
+Here is `redis-master-controller.yaml`:
+
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  labels:
+    name: redis-master
+  name: redis-master
```

Comment:
I won't comment individually on each of them, but all the .yaml reordering, and also their README listings, should now be done.